Olwa/dev

owalk opened the pull request:
I fixed the collision logic!!! Super excited about this one :)

Sean-Nishi commented on game/entities/entity.py:
```diff
@@ -391,6 +410,151 @@ def facing_towards_entity(self, other_sprite: pygame.sprite) -> bool:
+        else:
+            # top or bottom collision
+            if self.hitbox.centery < collision_rect.centery:
+                # teleport to the left
```
Comment should be teleport to up or down

owalk replied:
I fixed a couple of other comments in this method as well as the one you pointed out. Good catch

Sean-Nishi reviewed (changes requested):
All good except for the one comment not describing the right direction

owalk commented:
I found two additional bugs this morning. The self colliding needs a little wiggle room when being removed from inside a collision rect. fixed below:
![image](https://github.com/Sean-Nishi/Reluctant-Hero/assets/10580211/e4fbbedc-d81f-40a2-8b2e-2b94f213d18e)
Additionally, vector reflections were able to get stuck going back and forth even with this change, so I added additional checks for this previously uncaught edge case
![image](https://github.com/Sean-Nishi/Reluctant-Hero/assets/10580211/4dca5f40-b299-4864-a127-fbb0954404d0)

Sean-Nishi reviewed (approved):
Good job removing collision_direction in player's collision_set_compass. Using math vector2s is so much cleaner
